### PR TITLE
Call NewRule.setActivatedByDefault based on default profile

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: src/main/java/org/sonarsource/analyzer/commons/RuleMetadataLoader.java [hunk 1]
```diff
@@ -20,12 +20,15 @@
...
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
 import org.sonar.api.rule.RuleStatus;
 import org.sonar.api.rules.RuleType;
+import org.sonar.api.server.rule.RulesDefinition;
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/analyzer/commons/RuleMetadataLoader.java]: MINOR Remove this unused import 'org.sonar.api.server.rule.RulesDefinition'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/sonarsource/analyzer/commons/RuleMetadataLoader.java [hunk 2]
```diff
@@ -41,12 +43,22 @@
 
   private static final char RESOURCE_SEP = '/';
   private final String resourceFolder;
+  private final Set<String> activatedByDefault;
 
   private JsonParser jsonParser;
 
   public RuleMetadataLoader(String resourceFolder) {
```
Comment (vilchik-elena) [inline src/main/java/org/sonarsource/analyzer/commons/RuleMetadataLoader.java]: @pynicolas add doc explaining which constructor to use depending on SQ version

Comment (pynicolas) [inline src/main/java/org/sonarsource/analyzer/commons/RuleMetadataLoader.java]: @vilchik-elena I added a comment on the other constructor in ed53e25

**Pull request merged**